api.ts

```diff
@@ -187,10 +188,17 @@ export interface SourceFileConfiguration {
     readonly compilerPath?: string;
 
     /**
-     * Arguments for the compiler.
+     * Arguments for the compiler. These arguments should not require shell parsing, so should not include any
+     * shell variables, quoting or escaping.
      */
     readonly compilerArgs?: string[];
 
+    /**
+     * Command line fragments for the compiler. These are similar to compiler arguments, but support shell parsed
+     * content such as shell quoting and escaping.
+     */
+    readonly compilerFragments?: string[];

```

Comment:
Would it be better named something like compilerArgsShellProcessed?

Reply:
Also, what's the scenario in which this would be used? Is this just for backwards compatibility or do we know of specific cases?

Reply:
I updated the description to hopefully answer your questions.

CMake provides command line fragments, which CMake Tools processes.  This would allow CMake Tools to pass those fragments directly to cpptools, where we have code to do the conversion accurately (in one place).

Reply:
I don't see an updated description.

Reply:
I meant the PR description.  Can you suggest better descriptions for the fields?

Reply:
I don't know. I guess it's fine.